.github/workflows/ci.yml

```diff
@@ -123,24 +151,27 @@ jobs:
           path: |
             ~/go/pkg/mod
             ~/.cache/go-build
-          key: cache-go-${{ env.go-version }}-unittests-${{ hashFiles('**/go.sum') }}
+          key: ${{ runner.os }}-cache-go-${{ env.go-version }}-unittests-${{ hashFiles('**/go.sum') }}
           restore-keys: |
-            cache-go-${{ env.go-version }}-unittests-
-            cache-go-${{ env.go-version }}-
+            ${{ runner.os }}-cache-go-${{ env.go-version }}-unittests-
+            ${{ runner.os }}-cache-go-${{ env.go-version }}-
       - name: setup env
         run: make
       - name: Clear test cache
         run: make clear-test-cache
       - name: unit tests
         env:
-          GOTESTSUM_FORMAT: testname
+          GOTESTSUM_FORMAT: standard-verbose
           GOTESTSUM_JUNITFILE: unit-tests.xml
         run: make test
+        continue-on-error: ${{ matrix.allow-failure }}
       - name: Publish Test Report
         uses: mikepenz/action-junit-report@v3
-        if: always() # always run even if the previous step fails
+        # always run even if the previous step fails
+        if: ${{ always() && matrix.os == 'ubuntu-latest' }}
```

Comment:
Can't it be run on all platforms? :(

Reply:
It can but the `mikepenz/action-junit-report@v3` marks the build as failed if the report contains a failing test. The alternative to this would be to do something like:

```yaml
if: always()
with:
  report_paths: "**/unit-tests.xml"
  fail_on_failure: ${{ ! matrix.allow-failure }}
```